scripts/variantstore/wdl/GvsBulkIngestGenomes.wdl

```diff
@@ -98,105 +111,109 @@ workflow GvsBulkIngestGenomes {
+    meta {
+        volatile: true # always run this when asked otherwise you can get a previously run workspace
+    }
+    command <<<
+        # Prepend date, time and pwd to xtrace log entries.
+        PS4='\D{+%F %T} \w $ '
+        set -o errexit -o nounset -o pipefail -o xtrace
+
+        # Sniff the workspace bucket out of the delocalization script and extract the workspace id from that.
+        sed -n -E 's!.*gs://fc-(secure-)?([^\/]+).*!\2!p' /cromwell_root/gcs_delocalization.sh | sort -u > workspace_id.txt
+        sed -n -E 's!.*gs://(fc-(secure-)?[^\/]+).*!\1!p' /cromwell_root/gcs_delocalization.sh | sort -u > workspace_bucket.txt
+    >>>
+
+    runtime {
+        docker: "ubuntu:latest"
     }
 
-    task GetWorkspaceName {
+    output {
+        String workspace_id = read_string("workspace_id.txt")
+        String workspace_bucket = read_string("workspace_bucket.txt")
+    }
+}
 
-        input {
-            String workspace_id
-            String workspace_bucket
-        }
+task GetWorkspaceName {
+    ## In order to get the name and namespace of the workspace, without requiring that the user input it manually, we hit rawls directly for the exact name
+    input {
+        String workspace_id
+        String workspace_bucket
+    }
 
-        String workspace_name_output = "workspace_name.txt"
-        String workspace_namespace_output = "workspace_namespace.txt"
+    String workspace_name_output = "workspace_name.txt"
+    String workspace_namespace_output = "workspace_namespace.txt"
 
-        command <<<
-            # Hit rawls with the workspace ID
+    command <<<
+        # Hit rawls with the workspace ID
 
-            export WORKSPACE_BUCKET='~{workspace_bucket}'
+        export WORKSPACE_BUCKET='~{workspace_bucket}'
 
-            python3 /app/get_workspace_name_for_import.py \
-              --workspace_id ~{workspace_id} \
-              --workspace_name_output ~{workspace_name_output} \
-              --workspace_namespace_output ~{workspace_namespace_output} \
+        python3 /app/get_workspace_name_for_import.py \
+        --workspace_id ~{workspace_id} \
+        --workspace_name_output ~{workspace_name_output} \
+        --workspace_namespace_output ~{workspace_namespace_output} \
 
-        >>>
-        runtime {
-            docker: "us.gcr.io/broad-dsde-methods/variantstore:2023-05-08-alpine"
-            memory: "3 GB"
-            disks: "local-disk 10 HDD"
-            cpu: 1
-        }
+    >>>
+    runtime {
+        docker: "us.gcr.io/broad-dsde-methods/variantstore:2023-06-06-rc-21366d4ba5b5d28f3986123279c4637d1e88180f"
+        memory: "3 GB"
+        disks: "local-disk 10 HDD"
+        cpu: 1
+    }
 
     output {
         String workspace_name = read_string(workspace_name_output)
         String workspace_namespace = read_string(workspace_namespace_output)
     }
 }
 
+task GetColumnNames {
```

Comment:
I think this task should fail with a useful error if it can't return a column name. I ran a [workflow](https://app.terra.bio/#workspaces/gvs-dev/GVS%20Exome%20Data%2049k/job_history/ab1fdab5-819f-4089-99a3-1112ace19e8a) where I referenced a sample set that pointed to a sample table that had gVCFs in different columns (that is non-controls in one column name, controls in a different column name). This task returned an empty string for `vcf_files_column_name` (and then index). And the *next* task failed with a more confusing error. Would be easier to figure out if this task failed and said something helpful ("Couldn't determine unique column name???")

Reply:
ok